src/app/components/contact-info/contact-info.js

```diff
@@ -1,43 +1,93 @@
+    const [value, setValue] = useState('');
+    const [schoolLocation, setSchoolLocation] = useState();
+    const schoolIsOk = schoolSet.has(value.toLowerCase());
+    const selectedEntry = schoolIsOk && schools.find((s) => s.name.toLowerCase() === value.toLowerCase());
+    const schoolTypeOptions = React.useMemo(
+        () => schools.reduce((a, b) => {
+            if (b.type && !a.includes(b.type)) {
+                a.unshift(b.type);
+            }
+            return a;
+        }, ['Other']).map((text) => ({label: text, value: text})),
+        [schools]
+    );
 
-    React.useEffect(() => {
-        shellBus.emit(showPopup ? 'with-modal' : 'no-modal');
+    function onChange({target}) {
+        setValue(target.value);
+    }
 
-        return () => shellBus.emit('no-modal');
-    }, [showPopup]);
+    return (
+        <React.Fragment>
+            <FormInput
+                label="School name"
+                suggestions={schoolNames}
+                inputProps={{
+                    type: 'text',
+                    name: 'school',
+                    required: true,
+                    autocomplete: 'off',
+                    value,
+                    onChange
+                }}
+            />
+            {
+                selectedEntry &&
+                <React.Fragment>
+                    <input type="hidden" name="school_type" value={selectedEntry.type} />
+                    <input type="hidden" name="school_location" value={selectedEntry.location} />
+                    <input
+                        type="hidden" name="school_total_enrollment"
+                        value={selectedEntry.total_school_enrollment}
+                    />
+                </React.Fragment>
+            }
+            {
+                !schoolIsOk && value.length > 3 &&
+                <div className="school-info">
+                    <div className="instructions">
+                        We don&apos;t have that school in our system. Please tell us a little
+                        more about it:
+                    </div>
+                    <FormSelect
+                        label="What type of school is it?"
+                        name="school_type"
+                        selectAttributes={{required: true}}
+                        options={schoolTypeOptions}
+                    />
+                    <FormRadioGroup
+                        longLabel="Is it located in the United States?"
+                        name="school_location"
+                        options={[
+                            {label: 'Yes', value: 'Domestic'},
+                            {label: 'No', value: 'Foreign'}
+                        ]}
+                        selectedValue={schoolLocation}
+                        setSelectedValue={setSchoolLocation}
+                        required
+                    />
+                </div>
+            }
+        </React.Fragment>
+    );
+}
 
```

Comment:
it's a matter of preference but I dislike huge conditionally rendered blocks, because it's too hard to tell what gets rendered.  I find it cleaner to break each block out into a component and either pass a `isHidden` prop or have it contain the rendering logic internally.  Either way it can simply `return null` when it shouldn't render.